### PR TITLE
Fetch and Save Gemfiles For Security Auditing

### PR DESCRIPTION
**Pull request by emileswarts**
Fetch all the Gemfiles for applications.  Save them in tmp, where the scanner can do the auditing.

https://trello.com/c/1ZuyqP0q/93-investigate-integrating-github-security-vulnerability-alerts

### CODE REVIEW

File: lib/gateways/gemfile.rb [hunk 1]
```diff
@@ -3,8 +3,12 @@
 module Gateways
   class Gemfile
     def execute(application_name:)
-      open(gemfile_url(application_name)) do |gemfile_contents|
-        Domain::Gemfile.new(file_contents: gemfile_contents.read)
+      begin
+        open(gemfile_url(application_name)) do |gemfile_contents|
+          Domain::Gemfile.new(file_contents: gemfile_contents.read)
+        end
+      rescue(OpenURI::HTTPError) => _
```
Comment (danielburnley) [inline lib/gateways/gemfile.rb]: Feels kinda weird it fails silently, feels like the usecase could be the one rescuing from this exception, and here we just throw our own one like `GemfileNotFoundException` or something. Not sure what the usual approach is

File: spec/use_cases/fetch_gemfiles_spec.rb [hunk 2]
```diff
@@ -52,4 +63,17 @@
       ])
     end
   end
+
+  context 'The gateway raised a GemfileNotFoundException' do
```
Comment (danielburnley) [inline spec/use_cases/fetch_gemfiles_spec.rb]: Should also test that the result is empty so you can see that's the expected behaviour from reading the tests
